server/src/main/kotlin/org/javacs/kt/Configuration.kt

```diff
@@ -35,6 +48,39 @@ public data class ExternalSourcesConfiguration(
     var autoConvertToKotlin: Boolean = false
 )
 
+
+fun setupServerDatabase(params: InitializeParams): Database? {
+    val dbName = "kls_database"
+
+    params.initializationOptions?.let { initializationOptions ->
+        val gson = GsonBuilder().registerTypeHierarchyAdapter(Path::class.java, GsonPathConverter()).create()
+        val options = gson.fromJson(initializationOptions as JsonElement, InitializationOptions::class.java)
+
+        options.storagePath?.let { storagePath ->
+            if (Files.isDirectory(storagePath)) {
+                return Database.connect("jdbc:sqlite:${Path.of(storagePath.toString(), dbName)}.db")
+            }
+        }
+    }
+
+    return null
+}
```

Comment:
I feel like this should be moved somewhere other than `Configuration`, ideally where the `Database` connection lives. Thinking about this, perhaps we'd want to create a wrapper class around the `Database` that deals with this lazy setup internally? That way, we could pass the wrapper directly in the constructor to `CompilerClassPath` and `SourcePath`, wouldn't need the `lateinit` and could also make it `db` private in `CompilerClassPath`.

Reply:
Can you maybe elaborate on what the wrapper class would look like? Is something like this what you had in mind?

```kotlin
class DatabaseService {

    var db: Database? = null
        private set

    fun setupDatabase(storagePath: Path?) {
        val dbName = "kls_database"

        db = if (Files.isDirectory(storagePath)) {
            Database.connect("jdbc:sqlite:${Path.of(storagePath.toString(), dbName)}.db")
        } else {
            null
        }
    }
}
```

And then `CompilerClassPath` and `SourcePath` would receive the `DatabaseService`. And when they need to use the db they would just call `databaseService.db`.

`databaseService.setupDatabase(storagePath)` would be called on `initialize`.

Reply:
Yes, that's pretty close to what I had in mind. I'd shorten the function name to `setup`, since the wrapper should also make it clear that we are setting up the database, but otherwise that looks good 👍